app/models/service_announcement.rb

```diff
@@ -1,2 +1,12 @@
 class ServiceAnnouncement < ApplicationRecord
+  validates :color, inclusion: {
+    in: %w( plain red orange yellow olive green teal blue violet purple pink brown grey black )
+  }
+
+  default_scope { active }
```

Comment:
You have this default scope but you never use it. I think your code will be cleaner without it.

Reply:
I do use it :D `ServiceAnnouncement.all` is now scoped to only the active and current announcements. But I did notice it was getting to be a pain in the admin panel where everything has to be `.unscoped` so I might take off the default scope bit.